### DISCUSSION TITLE
passing accept-routes via terminal

### DISCUSSION THREAD

**Discussion by abuschel** [hassio-addons/app-tailscale#408] [Ideas]
I need to connect HA to remote cameras, but I cannot run the tailscale command and pass the --accept-routes switch. Is there a reason why?

**Comment by lmagyar** [comment]
Did you read the DOCS? See `accept_routes`. This is enabled by default.

Do your cameras have Tailscale on them? Or there is a subnet router for them? Their local IPs will be source-NAT-ed on the other Tailscale location, no need to modify anything. Did you enable your cameras' subnet under "Edit route settings..." on this site: https://login.tailscale.com/admin/machines ?

**Comment by abuschel** [comment]
Thank you for the reply.

I run a terminal on HA .

I cannot run the Tailscale command to do a Tailscale ping.

The cameras do not run Tailscale, but there is a machine that does and it
advertises the subnet where the cameras are connected.

HA ip is 192.168.1.152
Camera IP is 192.168.128.100
Configuration.yaml is correct and points to camera

I can reach HA IP from 192.168.128.2 with no issues



____________________________
Alejandro (Alex) Buschel
Managing Partner
ProGloBix
703-996-9187
***@***.***
Signal:+14252418834
Telegram: @alexgb1


On Fri, Sep 20, 2024 at 09:26 Laszlo Magyar ***@***.***>
wrote:

> Did you read the DOCS? See accept_routes. This is enabled by default.
>
> Do your cameras have Tailscale on them? Or there is a subnet router for
> them? Their local IPs will be source-NAT-ed on the other Tailscale
> location, no need to modify anything. Did you enable your cameras' subnet
> under "Edit route settings..." on this site:
> https://login.tailscale.com/admin/machines ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hassio-addons/addon-tailscale/discussions/408#discussioncomment-10704704>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEJ5MVUFEMXD42LUMVO74GDZXQIA3AVCNFSM6AAAAABOQ7NDJWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANZQGQ3TANA>
> .
> You are receiving this because you authored the thread.Message ID:
> <hassio-addons/addon-tailscale/repo-discussions/408/comments/10704704@
> github.com>
>

-- 


This
 email and any files transmitted with it are confidential and 
intended 
solely for the use of the individual or entity to whom they are 
addressed. If you have received this email in error please notify the 
system manager. This message contains confidential information and is 
intended only for the individual named. If you are not the named 
addressee 
you should not disseminate, distribute or copy this e-mail. 
Please notify 
the sender immediately by e-mail if you have received this
 e-mail by 
mistake and delete this e-mail from your system. If you are 
not the 
intended recipient you are notified that disclosing, copying, 
distributing 
or taking any action in reliance on the contents of this 
information is 
strictly prohibited.

**Comment by lmagyar** [comment]
Did you approve the subnet route (192.168.128.x) on the admin web page of the 192.168.128.2 device (I assume TS for the camera subent is running on this machine)?

Did you disable userspace networking in the add-on options? Better done in YAML, the UI is broken, it shows "off" even when it is not configured (instead of grayed out), and when not configured it is enabled by default.

With these you should access from HA on 192.168.1.152 (LAN1) the devices behind subnet router 192.168.128.2 (LAN2) on subnet 192.168.128.x.

**Comment by lmagyar** [comment]
> I cannot run the Tailscale command to do a Tailscale ping

Yeah, that terminal is the terminal add-on's terminal, and that is a different Docker container than the Tailscale add-on's container. You can access the TS add-on's terminal, but you should not. When userspace_networking is disabled, there will be a tailscale0 interface on the host (and for all containers), and HA (in another container) will access the tailnet (and the subnet) through this tailscale0, everything is configured by TS automatically.

**Comment by abuschel** [comment]
Thank you Laszlo for your help. I made the change you recommended to
userspace_networking and now everything works.

It would be great to be able to do a tailscale ping from a terminal on HA
to check who is providing a specific route.

Alex

____________________________
Alejandro (Alex) Buschel
Managing Partner
ProGloBix
703-996-9187
***@***.***
Signal:+14252418834
Telegram: @alexgb1


On Fri, Sep 20, 2024 at 1:22 PM Laszlo Magyar ***@***.***>
wrote:

> I cannot run the Tailscale command to do a Tailscale ping
>
> Yeah, that terminal is the terminal add-on's terminal, and that is a
> different Docker container than the Tailscale add-on's container. You can
> access the TS add-on's terminal, but you should not. When
> userspace_networking is disabled, there will be a tailscale0 interface on
> the host (and for all containers), and HA (in another container) will
> access the tailnet (and the subnet) through this tailscale0, everything is
> configured by TS automatically.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hassio-addons/addon-tailscale/discussions/408#discussioncomment-10709114>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEJ5MVUZCAS3JTIAAPR7O63ZXR7WZAVCNFSM6AAAAABOQ7NDJWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANZQHEYTCNA>
> .
> You are receiving this because you authored the thread.Message ID:
> <hassio-addons/addon-tailscale/repo-discussions/408/comments/10709114@
> github.com>
>

-- 


This
 email and any files transmitted with it are confidential and 
intended 
solely for the use of the individual or entity to whom they are 
addressed. If you have received this email in error please notify the 
system manager. This message contains confidential information and is 
intended only for the individual named. If you are not the named 
addressee 
you should not disseminate, distribute or copy this e-mail. 
Please notify 
the sender immediately by e-mail if you have received this
 e-mail by 
mistake and delete this e-mail from your system. If you are 
not the 
intended recipient you are notified that disclosing, copying, 
distributing 
or taking any action in reliance on the contents of this 
information is 
strictly prohibited.

**Comment by lmagyar** [comment]
Yeah, you are right, that is a valid point to access the TS add-on's terminal. You can do it by:
- access the Terminal add-on's teminal
- ```docker exec -it `docker ps -q -f name=tailscale` /bin/bash```
- `/opt/tailscale ping your-other-device`

If you see a relay even after several ping, ie. it can't find a direct route, try to set the new UDP port networking config option and if possible set up port forwarding on the router. This way TS can find a direct route for CG-NAT-ed devices (ie. phones). TS should find a route without this, but in practice sometimes it fails (I don't know why) and uses relays.

And please remember, not to configure TS through it's commandline, it will interfere with the add-on's startup logic. And don't update it, the container will revert on the next startup. The add-on's command line is not for basic/general configuration, it is for diagnostics/development.

**Comment by abuschel** [comment]
Hi,


I have turned on Funnel, after updating to the latest version of the add-on.

The behavior now is consistent, but different. The add-on seems to be
running, but it is not available via the menu on the navbar on the left,
see below

[image: image.png]
[image: image.png]
[image: image.png]
 I have turned on logging, can provide it if needed.

Alex

____________________________
Alejandro (Alex) Buschel
Managing Partner
ProGloBix
703-996-9187
***@***.***
Signal:+14252418834
Telegram: @alexgb1


On Wed, Sep 25, 2024 at 9:47 AM Laszlo Magyar ***@***.***>
wrote:

> Yeah, you are right, that is a valid point to access the TS add-on's
> terminal. You can do it by:
>
>    - access the Terminal add-on's teminal
>    - docker exec -it `docker ps -q -f name=tailscale` /bin/bash
>    - /opt/tailscale ping your-other-device
>
> If you see a relay even after several ping, ie. it can't find a direct
> route, try to set the new UDP port networking config option and if possible
> set up port forwarding on the router. This way TS can find a direct route
> for CG-NAT-ed devices (ie. phones). TS should find a route without this,
> but in practice sometimes it fails (I don't know why) and uses relays.
>
> And please remember, not to configure TS through it's commandline, it will
> interfere with the add-on's startup logic. And don't update it, the
> container will revert on the next startup. The add-on's command line is not
> for basic/general configuration, it is for diagnostics/development.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hassio-addons/addon-tailscale/discussions/408#discussioncomment-10753998>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEJ5MVSUINQGPNKQZEZ4TJ3ZYLSJPAVCNFSM6AAAAABOQ7NDJWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANZVGM4TSOA>
> .
> You are receiving this because you authored the thread.Message ID:
> <hassio-addons/addon-tailscale/repo-discussions/408/comments/10753998@
> github.com>
>

-- 


This
 email and any files transmitted with it are confidential and 
intended 
solely for the use of the individual or entity to whom they are 
addressed. If you have received this email in error please notify the 
system manager. This message contains confidential information and is 
intended only for the individual named. If you are not the named 
addressee 
you should not disseminate, distribute or copy this e-mail. 
Please notify 
the sender immediately by e-mail if you have received this
 e-mail by 
mistake and delete this e-mail from your system. If you are 
not the 
intended recipient you are notified that disclosing, copying, 
distributing 
or taking any action in reliance on the contents of this 
information is 
strictly prohibited.

**Reply by lmagyar** [reply]
I can't see any image, I think you should use the web UI of Github and not email replies, but it is only a guess.
